observability/WFLY-14854_add_opentelemetry_support.adoc

```diff
@@ -0,0 +1,76 @@
+ = Add OpenTelemetry Tracing Support to WildFly
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+https://opentelemetry.io[OpenTelemetry] is an ambitious specification that aims to encapsulate APIs for tracing, metrics, and logging. With OpenTracing (which WildFly already supports) having shutdown in favor of OpenTelemetry, we need to migrate our support as well. This project will only cover the tracing aspects of OpenTelemetry. Any future support for the metrics and logging parts of the OpenTelemetry spec will need to be handled by a separate effort.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14854
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1688
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+* [X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The server/domain administrator should be able to configure OpenTelemetry export and span processor options
+* Jakarta REST endpoints should be automatically traced
```

Comment:
All of the application's Jakarta REST endpoints would be traced by default? Can the tracing be turned off/configured to match some pattern?

Reply:
Yes, all will be traced by default. Turning that off by pattern is certainly something can be added at some point, but it is not supported at present.

Reply:
Is it something opentracing provide today?

Reply:
It is not something we offer. There are ways to do it, and we have open RFEs to implement that, but we do not currently support that, so it's not a loss of functionality.

Reply:
Yes, REST endpoints are currently automatically traced if OpenTracing is enabled.